### PR TITLE
feat: Avoid storing file history for API based wfs

### PR DESCRIPTION
**Pull request by chandrasekharan-zipstack**
## What

- Set and passed `use_file_history` for WF execution in case of API based source
- Included changes for v2 migration as well
- Allows File / DB based runs to still cache the file history (can be reviewed later), API based runs are not stored
- Set API deployment timeout's min value and default to 0 instead of -1

## Why

- To avoid caching file history for API based WF runs

## Can this PR break any existing features. If yes, please list possible items. If no, please explain why. (PS: Admins do not merge the PR without this section filled)

- No, added params with sane defaults

## Related Issues or PRs

- #745 

## Dependencies Versions

-

## Notes on Testing

- Attempted running an API based WF locally

## Checklist

I have read and understood the [Contribution Guidelines]().

### CODE REVIEW

**Comment by chandrasekharan-zipstack** [comment]
Code duplication is due to the v2 migration changes also being present in this PR

File: backend/workflow_manager/endpoint/source.py [hunk 1]
```diff
@@ -329,7 +329,11 @@ def _is_new_file(self, file_path: str, workflow: Workflow) -> bool:
             workflow=workflow, cache_key=file_hash
         )
 
-        if file_history and file_history.is_completed():
+        if (
+            self.execution_service.use_file_history
+            and file_history
+            and file_history.is_completed()
+        ):
```
Comment (ritwik-g) [inline backend/workflow_manager/endpoint/source.py:336]: @chandrasekharan-zipstack could you please explain what exactly are we doing here? May be add a comment in the code also?

File: backend/api/serializers.py [hunk 2]
```diff
@@ -85,15 +85,15 @@ class ExecutionRequestSerializer(Serializer):
 
     Attributes:
         timeout (int): Timeout for the API deployment, maximum value can be 300s.
-            If -1 it corresponds to async execution. Defaults to -1
+            If 0 it corresponds to async execution. Defaults to 0
```
Comment (ritwik-g) [inline backend/api/serializers.py:88]: @chandrasekharan-zipstack should the default be 0 or the maximum value?

Comment (chandrasekharan-zipstack) [reply backend/api/serializers.py:88]: @ritwik-g in the previous implementation it defaulted to async execution - I let it remain the same. If we're changing this behaviour we'd have to inform users this as a breaking change

Comment (ritwik-g) [reply backend/api/serializers.py:88]: @chandrasekharan-zipstack if that is the case then isn't removing `-1` also a breaking change? Should we then keep the min as `-1` and default as `-1`

Comment (chandrasekharan-zipstack) [reply backend/api/serializers.py:88]: @ritwik-g I wonder why I didn't catch that - let me revert the change if we're worried about this breaking change. We can make it later if necessary

**Review by ritwik-g — APPROVED** [review]

**Review by muhammad-ali-e — APPROVED** [review]
LGTM. Just to confirm, we're only saving the result in the database when the API is called with specific parameters, right?

**Comment by chandrasekharan-zipstack** [comment]
> LGTM. Just to confirm, we're only saving the result in the database when the API is called with specific parameters, right?

@muhammad-ali-e in case of workflow execution, we'll never save the results from API related source / destination. Only for ETL / task (file/ DB related source / destination) will make use of the file history table to mark processed files (not results)

**Pull request merged**